crates/services/compression/src/service.rs

```diff
@@ -0,0 +1,191 @@
+use crate::ports::{
+    block_source::BlockSource,
+    compression_storage::CompressionStorage,
+    configuration::CompressionConfigProvider,
+};
+use fuel_core_services::{
+    EmptyShared,
+    RunnableService,
+    RunnableTask,
+    ServiceRunner,
+    StateWatcher,
+};
+
+/// The compression service.
+/// Responsible for subscribing to the l2 block stream,
+/// perform compression of the block, and store the compressed block.
+/// We don't need to share any data between this task and anything else yet(?)
+/// perhaps we want to expose a way to get the registry root
+pub struct CompressionService<B, S, C> {
+    /// The block source.
+    block_source: B,
+    /// The compression storage.
+    storage: S,
+    /// The compression config.
+    config: C,
+}
+
+impl<B, S, C> CompressionService<B, S, C>
+where
+    B: BlockSource,
+    S: CompressionStorage,
+    C: CompressionConfigProvider,
+{
+    fn _new(block_source: B, storage: S, config: C) -> Self {
```

Comment:
I assume this should be public, in which case I don't think we'll get the unused warning.
```suggestion
    pub fn new(block_source: B, storage: S, config: C) -> Self {
```

Reply:
this is not meant to be public. we're wrapping it in new_service :)

Reply:
fair though